src/test/java/pl/widulinski/webcrawlertool/ScrAppApplicationTests.java

```diff
@@ -0,0 +1,157 @@
+package pl.widulinski.webcrawlertool;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.poi.openxml4j.exceptions.InvalidFormatException;
+import org.junit.jupiter.api.Test;
+import org.junit.runner.RunWith;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.chrome.ChromeOptions;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import pl.widulinski.webcrawlertool.FoundWebElements.FoundWebElement;
+import pl.widulinski.webcrawlertool.createExcelFiles.CreateExcel;
+import pl.widulinski.webcrawlertool.enums.Categories;
+import pl.widulinski.webcrawlertool.searchData.PreparedDataToSrchDto;
+import pl.widulinski.webcrawlertool.webDataToScrap.DataToScrap;
+import pl.widulinski.webcrawlertool.webDataToScrap.DataToScrapRepository;
+
+import java.io.File;
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.*;
+import java.util.stream.Stream;
+
+import static org.hamcrest.Matchers.containsString;
+import static org.junit.Assert.assertTrue;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+
+
+@RunWith(SpringRunner.class)
+@AutoConfigureMockMvc
+@Slf4j
+@SpringBootTest
+class ScrAppApplicationTests {
```

Comment:
pierwszy test nie testuje wcale kodu ktory napisales tylko tworzy nowy kod i sprawdza co tam się zadzieje. 
drugi test probuje przetestować wszysztko na raz. Przetestuj osobno pobieranie danych z bazy, osobno tworzenie pliku i osobno bieganie po stornie i wyciąganie danych.

Reply:
rozdzieliłem testy na osobne klasy- tak jak zrobił to Grzesiek.
testuje: endpointy, zapisywanie do pliku excel i skrobanie ze stron- używam tutaj klasy pomocniczej ExcelReader do odczytywania danych z pliku xlsx

Reply:
zapomniałem jeszcze o testowaniu zapisu do repo z klasty Dto